pkg/cloud/aws/aws.go

```diff
@@ -76,3 +77,32 @@ func NewProviderAssets(config config.OperatorConfig) (common.CloudProviderAssets
 	}
 	return assets, nil
 }
+
+// CloudConfigTransformer implements the cloudConfigTransformer. It uses the source ConfigMap and
+// infra.status.platformStatus.AWS.ServiceEndpoints to create a new config that include the
+// ServiceOverrides sections. The transformer uses infra.status.platformStatus.region as the
+// signing service for all the ServiceOverrides.
+// It returns an error if the platform is not AWSPlatformType.
+func CloudConfigTransformer(source string, infra *configv1.Infrastructure) (string, error) {
+	if !(
+		infra.Status.PlatformStatus != nil &&
+		infra.Status.PlatformStatus.Type == configv1.AWSPlatformType) {
+		return "", fmt.Errorf("invalid platform, expected to be %s", configv1.AWSPlatformType)
+	}
+	if infra.Status.PlatformStatus.AWS == nil || len(infra.Status.PlatformStatus.AWS.ServiceEndpoints) == 0 {
+		return source, nil  // no transformation required
+	}
+
+	var region string
+	if infra.Status.PlatformStatus != nil && infra.Status.PlatformStatus.AWS != nil {
+		region = infra.Status.PlatformStatus.AWS.Region
+	}
+	if region == "" {
+		// TODO: CCM used field.Required for this; do we want to add that dep?
```

Comment:
Which go module does the `field` package come from, if it's a standard K8s like apimachinery, then go ahead

Reply:
Yup, `k8s.io/apimachinery/pkg/util/validation/field`

Reply:
Seems fine to use it then, we already import apimachinery